java/src/main/java/ai/rapids/cudf/BaseDeviceMemoryBuffer.java

```diff
@@ -43,6 +43,22 @@ public final void copyFromHostBuffer(long destOffset, HostMemoryBuffer src, long
     Cuda.memcpy(address + destOffset, src.address + srcOffset, length, CudaMemcpyKind.HOST_TO_DEVICE);
   }
 
+  /**
+   * Copy a subset of src to this buffer starting at destOffset using the specified CUDA stream.
+   * @param destOffset the offset in this to start copying from.
+   * @param src what to copy from
+   * @param srcOffset offset into src to start out
+   * @param length how many bytes to copy
+   * @param stream CUDA stream to use
```

Comment:
Some more documentation on this would be really good, because this is opening up a whole new set of issues.  Please make sure that it is clear that the copy will have completed by the time this happens and the stream only makes it asynchronous to other copies/etc, so they don't need to worry about freeing the host memory buffer while a copy is still happening.

Reply:
Updated the comments to be more explicit on the methods synchronizing with host but not necessarily other streams.